contributors/design-proposals/bootstrap-discovery.md

```diff
@@ -173,6 +172,47 @@ In addition, `jws-kubeconfig-<token-id>-hash` will be set to the MD5 hash of the
 
 This controller will also delete tokens that are past their expiration time.
 
+## `kubeadm` UX
+
+We extend kubeadm with a set of flags and helper commands for managing and using these tokens.
+
+### `kubeadm init` flags
+
+* `--token` If set, this injects the bootstrap token to use when initializing the cluster.  If this is unset, then a random token is created and shown to the user.  If set explicitly to the empty string then no token is generated or created.  This token is used for both discovery and TLS bootstrap.
+* `--token-ttl` If set, this sets the TTL for the lifetime of this token.  Defaults to 0 which means "forever"
+
+### `kubeadm join` flags
+
+* `--token` This sets the token for both discovery and bootstrap auth.
+* `--discovery-url` If set this will grab the cluster-info data (a kubeconfig) from a URL. Due to the sensitive nature of this data, we will only support https URLs.  This also supports `username:password@host` syntax for doing HTTP auth.
+* `--discovery-file` If set, this will load the cluster-info from a file.
+* `--discovery-token` If set, (or set via `--token`) then we will be using the token scheme described above.
```

Comment:
Sorry if I was unclear last review (I was in a hurry), but I do think we should keep what you had in terms of exposing both `--discovery-token` and `--tls-bootstrap-token` and using `--token` as a shorthand for setting both.

This way, the only `--discovery-xxx` flag we have is token, because that's actually the only one that's gonna do discovery with the cluster-info ConfigMap, JWS validation and all that. I see this as two phases:

1. Unsecure HTTP GET call to the cluster-info ConfigMap for getting the CA and master list and validate the content we recieved using the token -- this is discovery and is only available for the token method
2. TLS Bootstrap -- do the CSR dance and connect to the API Server with whatever creds you have (existing client cert or token), but this is most often a token regardless of method -- this phase is available for file, url and token.

I propose to define this in the doc and refer to 1. as `discovery` and 2. as just TLS bootstrap.
In the future, 2. may also be passed to the kubelet: https://github.com/kubernetes/kubeadm/issues/173

So TL;DR; `--discovery-token` is used in 1. and `--tls-bootstrap-token` is used in 2. and `--token` sets both